ffcx/element_interface.py

```diff
@@ -99,334 +59,23 @@ def map_facet_points(points: npt.NDArray[np.float64], facet: int, cellname: str)
-        for d in self._entity_counts[:-1]:
-            dofs += [[0] * d]
-
-        dofs += [[self.dim]]
-        return dofs
-
-    @property
-    def entity_dofs(self) -> typing.List[typing.List[typing.List[int]]]:
-        """DOF numbers associated with each entity."""
-        start_dof = 0
-        entity_dofs = []
-        for i in self.num_entity_dofs:
-            dofs_list = []
-            for j in i:
-                dofs_list.append([start_dof + k for k in range(j)])
-                start_dof += j
-            entity_dofs.append(dofs_list)
-        return entity_dofs
-
-    @property
-    def num_entity_closure_dofs(self) -> typing.List[typing.List[int]]:
-        """Number of DOFs associated with the closure of each entity."""
-        return self.num_entity_dofs
-
-    @property
-    def entity_closure_dofs(self) -> typing.List[typing.List[typing.List[int]]]:
-        """DOF numbers associated with the closure of each entity."""
-        return self.entity_dofs
-
-    @property
-    def num_global_support_dofs(self) -> int:
-        """Get the number of global support DOFs."""
-        return 1
-
-    @property
-    def reference_topology(self) -> typing.List[typing.List[typing.List[int]]]:
-        """Topology of the reference element."""
-        raise NotImplementedError()
-
-    @property
-    def reference_geometry(self) -> npt.NDArray[np.float64]:
-        """Geometry of the reference element."""
-        raise NotImplementedError()
-
-    @property
-    def family_name(self) -> str:
-        """Family name of the element."""
-        return self._family_name
-
-    @property
-    def lagrange_variant(self) -> typing.Union[basix.LagrangeVariant, None]:
-        """Basix Lagrange variant used to initialise the element."""
-        return None
-
-    @property
-    def dpc_variant(self) -> typing.Union[basix.DPCVariant, None]:
-        """Basix DPC variant used to initialise the element."""
-        return None
-
-    @property
-    def element_family(self) -> typing.Union[basix.ElementFamily, None]:
-        """Basix element family used to initialise the element."""
-        return None
-
-    @property
-    def cell_type(self) -> basix.CellType:
-        """Basix cell type used to initialise the element."""
-        return basix.cell.string_to_type(self._cellname)
-
-    @property
-    def discontinuous(self) -> bool:
-        """True if the discontinuous version of the element is used."""
-        return False
+# TODO: remove this deprecated function
+def QuadratureElement(
+    cellname: str, value_shape: typing.Tuple[int, ...], scheme: typing.Optional[str] = None,
+    degree: typing.Optional[int] = None, points: typing.Optional[npt.NDArray[np.float64]] = None,
+    weights: typing.Optional[npt.NDArray[np.float64]] = None, mapname: str = "identity"
+) -> basix.ufl._ElementBase:
+    warnings.warn(
+        "ffcx.element_interface.QuadratureElement is deprecated and will be removed after December 2023. "
```

Comment:
If we agree with the date for removal give here, we should open an issue to remind us to actually delete these functions

Reply:
Done: https://github.com/FEniCS/ffcx/issues/605